### ISSUE TITLE
CA2000 should not be reported when returning object that from another method

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn-analyzers#2361]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2
Note that this issue doesn't occur v2.9.1

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Repro steps
```cs
using System.IO;

class C
{
    Stream M() => File.OpenRead("C:/somewhere/");
}
```

#### Expected behavior
CA2000 is not reported. This method is intended to return an object.

#### Actual behavior
CA2000 is reported:
`Call System.IDisposable.Dispose on object created by 'File.OpenRead("C:/somewhere/")' before all references to it are out of scope.`
